core/src/test/java/com/mongodb/hadoop/splitter/StandaloneMongoSplitterTest.java

```diff
@@ -1,37 +1,51 @@
 package com.mongodb.hadoop.splitter;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.net.UnknownHostException;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
 import com.mongodb.BasicDBObject;
+import com.mongodb.BasicDBObjectBuilder;
 import com.mongodb.DBCollection;
+import com.mongodb.DBObject;
 import com.mongodb.MongoClient;
 import com.mongodb.MongoClientURI;
 import com.mongodb.hadoop.input.MongoInputSplit;
 import com.mongodb.hadoop.util.MongoClientURIBuilder;
 import com.mongodb.hadoop.util.MongoConfigUtil;
-import org.apache.hadoop.conf.Configuration;
-import org.apache.hadoop.mapreduce.InputSplit;
-import org.junit.Test;
-
-import java.net.UnknownHostException;
-import java.util.List;
-
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
 
 public class StandaloneMongoSplitterTest {
 
-    @Test
-    public void unshardedCollection() throws UnknownHostException, SplitFailedException {
+    private static MongoClientURI uri;
+
+    
+    @BeforeClass
+    public static void setUp() {
         MongoClient client = new MongoClient("localhost", 27017);
-        MongoClientURI uri = new MongoClientURIBuilder()
-                                 .collection("mongo_hadoop", "splitter_test")
-                                 .build();
+        uri = new MongoClientURIBuilder()
+            .collection("mongo_hadoop", "splitter_test")
+            .build();
         DBCollection collection = client.getDB(uri.getDatabase()).getCollection(uri.getCollection());
         collection.drop();
-        for (int i = 0; i < 10000; i++) {
+        collection.createIndex("value");
+        for (int i = 0; i < 40000; i++) {
             collection.insert(new BasicDBObject("_id", i)
-                                  .append("value", i)
-                             );
+                .append("value", i)
+                );
         }
+    }
+
+    @Test
+    public void unshardedCollection() throws UnknownHostException, SplitFailedException {
```

Comment:
l reused the collection creation and make it insert more values with an index. More rows were needed in order to properly test the min/max splitting. I included an index over a value in order to test splitting over something different than the _id.  and made the setup to save time during the tests. 
I could reorder the imports, it is eclipse annoying me.